run_unittests.py

```diff
@@ -4989,12 +4989,18 @@ def _check_ld(self, name: str, lang: str, expected: str) -> None:
                 self.assertEqual(comp.linker.id, expected)
 
     def test_link_environment_variable_lld_link(self):
+        if 'mingw' in platform.system().lower():
+            raise unittest.SkipTest('Cannot use VisualStudio-style compiler with GCC')
```

Comment:
I think it would be better to do:
```python
env = get_fake_env()
comp = env.detect_c_compiler(MachineChoice.HOST)
if comp.id == 'gcc':
    raise unittest.SkipTest('GCC cannot be used with link compatible linkers.')
```

Also note that your error has a typo "compiler" -> "linker"
```